src/ur5e_arm.cpp

```diff
@@ -402,15 +402,30 @@ void UR5eArm::send_trajectory(const std::vector<vector6d_t>& p_p,
 }
 
 // helper function to read a data packet and send a noop message
-void UR5eArm::read_and_noop() {
+bool UR5eArm::read_joint_keep_alive() {
     std::unique_ptr<rtde_interface::DataPackage> data_pkg = driver->getDataPackage();
     if (data_pkg) {
         // read current joint positions from robot data
         if (!data_pkg->getData("actual_q", joint_state)) {
-            throw std::runtime_error("couldn't get joint positions");
+            BOOST_LOG_TRIVIAL(error) << "read_joint_keep_alive driver->getDataPackage()->data_pkg->getData(\"actual_q\") returned false";
+            return false;
         }
 
         // send a noop to keep the connection alive
-        driver->writeTrajectoryControlMessage(control::TrajectoryControlMessage::TRAJECTORY_NOOP, 0, RobotReceiveTimeout::off());
+        if (!driver->writeTrajectoryControlMessage(control::TrajectoryControlMessage::TRAJECTORY_NOOP, 0, RobotReceiveTimeout::off())) {
+            BOOST_LOG_TRIVIAL(error) << "read_joint_keep_alive driver->writeTrajectoryControlMessage returned false";
+            return false;
+        };
+    } else {
+        // we received no data packet, so our comms are down. reset the comms from the driver.
+        BOOST_LOG_TRIVIAL(info) << "no packet found, resetting RTDE client connection";
+        try {
+            driver->resetRTDEClient(path_offset + OUTPUT_RECIPE, path_offset + INPUT_RECIPE);
```

Comment:
Can we please test:
1. Start up the ur5e module with a healthy conection with the arm
2. Confirm reads & writes work
3. unplug the computer running the ur5e module from the arm
4. run reads and writes
5. confirm the module doesn't crash and that the error messages returned are clearly due to failed connectivity between the module and the arm
6. plug the ethernet back in
7. run reads & writes
8. confirm the arm works as normal

If any of the above fail (especially the error messages being unclear) I recommend tracking connectivity with the arm through an atomic and early returning from GRPC methods with a connectivity error if connectivity is not working.